### PR TITLE
sceMpegAvcCsc: added range validity tests.

### PR DESCRIPTION
**Pull request by sum2012**
http://code.google.com/p/jpcsp/source/detail?r=3715

### CODE REVIEW

**Comment by sum2012** [comment]
I don't know the translation of this code
 if (rangeX + rangeWidth > width || rangeY + rangeHeight > height) {

**Comment by hrydgard** [comment]
hm, not much to translate?

    if (x + width > psmfWidth || y + height > psmfHeight)

where you'd have to go get psmfWidth and psmfHeight the same way JPCSP does.

Do you know what this fixes, if anything? Still worth being accurate of course, just wondering.

**Comment by sum2012** [comment]
I don't know translate "Screen.width "

**Comment by hrydgard** [comment]
Hm, that code.google.com link doesn't work anymore :(

**Comment by DragonNeos** [comment]
@hrydgard Use this link to view Jpcsp r3715 (or 24a0754): https://github.com/jpcsp/jpcsp/commit/24a07541ace67f82cb8f04194e7c6bc68fa64f8a

**Comment by unknownbrackets** [comment]
I tried to create a test for this, but I don't know how to use sceMpegAvcCsc.... I tried sceMpegAvcDecodeYCbCr, but it also fails.  I guess I need the right sort of video?

-[Unknown]

**Comment by hrydgard** [comment]
Does this actually fix any games?

File: Core/HLE/sceMpeg.cpp [hunk 1]
```diff
@@ -1851,15 +1851,21 @@ static u32 sceMpegAvcCsc(u32 mpeg, u32 sourceAddr, u32 rangeAddr, int frameWidth
 		WARN_LOG(ME, "sceMpegAvcCsc(%08x, %08x, %08x, %i, %08x): bad mpeg handle", mpeg, sourceAddr, rangeAddr, frameWidth, destAddr);
 		return -1;
 	}
-
-	DEBUG_LOG(ME, "sceMpegAvcCsc(%08x, %08x, %08x, %i, %08x)", mpeg, sourceAddr, rangeAddr, frameWidth, destAddr);
```
Comment (zllIJudgeIllz) [inline Core/HLE/sceMpeg.cpp]: image

File: Core/HLE/sceMpeg.cpp [hunk 2]
```diff
@@ -1851,15 +1851,21 @@ static u32 sceMpegAvcCsc(u32 mpeg, u32 sourceAddr, u32 rangeAddr, int frameWidth
...
 	int y = Memory::Read_U32(rangeAddr + 4);
 	int width = Memory::Read_U32(rangeAddr + 8);
 	int height = Memory::Read_U32(rangeAddr + 12);
-	int destSize = ctx->mediaengine->writeVideoImageWithRange(destAddr, frameWidth, ctx->videoPixelMode, x, y, width, height);
+	if (((x | y | width | height) & 0xF) != 0) {
+		WARN_LOG(ME, "sceMpegAvcCsc(%08x, %08x, %08x, %i, %08x) returning ERROR_MPEG_INVALID_VALUE", mpeg, sourceAddr, rangeAddr, frameWidth, destAddr);		
+		return ERROR_MPEG_INVALID_VALUE;
+	}
+	if (x < 0 || y < 0 || width < 0 || height < 0) {
+		WARN_LOG(ME, "sceMpegAvcCsc(%08x, %08x, %08x, %i, %08x) returning ERROR_INVALID_VALUE", mpeg, sourceAddr, rangeAddr, frameWidth, destAddr);
```
Comment (zllIJudgeIllz) [inline Core/HLE/sceMpeg.cpp]: image

**Comment by hrydgard** [comment]
Went ahead and rebased and merged this after 5 years. Still not sure what games are helped by this?

**Pull request closed without merging**